src/snowflake/snowpark/dataframe.py

```diff
@@ -1942,21 +1939,20 @@ def join(
             rsuffix: Suffix to add to the overlapping columns of the right DataFrame.
 
         Note:
-            If both ``lsuffix`` and ``rsuffix`` are empty, the overlapping columns will have random column names in the result DataFrame.
-            If either one is not empty, the overlapping columns won't have random names.
-
+            When both ``lsuffix`` and ``rsuffix`` are empty, the overlapping columns will have random column names in the resulting DataFrame.
+            You can reference to these randomly named columns using Column.alias (See the first usage in Examples).
```

Comment:
We should cross-reference the method `Column.alias`, I want this to be a link in the generated doc. Working on the update.